### PR TITLE
Fix Thrift compact protocol parser to handle unknown fields

### PR DESCRIPTION
**Pull request by unexge**
Fixes parsing of dict-page-offset-zero.parquet and other files that may contain Thrift metadata fields not present in the current schema definition.

The parser was failing with UnexpectedList when encountering fields in the Thrift stream that weren't expected in the Zig struct (either beyond max_field_id or not matching any defined field). This caused field data misalignment and parsing errors.

Changes:
- Add FieldType enum within StructReader for type-safe field handling
- Implement skipFieldData() to properly skip unknown field data based on type
- Track which fields are handled and skip unhandled fields
- Skip fields with field_id > max_field_id
- Add conformance test for dict-page-offset-zero.parquet (39 rows, all 1552)
- Update TESTING.md to mark dict-page-offset-zero.parquet as passing

This improves forward compatibility by allowing parzig to parse Parquet files written by newer library versions with additional metadata fields.

### CODE REVIEW

File: src/thrift/protocol/compact.zig [hunk 1]
```diff
@@ -198,6 +198,58 @@ pub fn StructReader(comptime T: type) type {
         break :blk field_names;
     };
 
+    fn skipFieldData(field_type_id: u4, reader_ptr: *Reader, arena: std.mem.Allocator) !void {
```
Comment (unexge) [inline src/thrift/protocol/compact.zig]: @claude can you fix the failing test?
